File: algorithm/Sorting-C/bubble_sort/bubble_sort.c
#include "bubble.h"

void bubble_sort(int list[], int n) {
    int i, j, temp;

    for (i = 0; i < n - 1; i++) {
        for (j = 0; j < n - 1; j++) {
            if (list[j] > list[j + 1]) {
                swap(&list[j], &list[j + 1]);
            }
        }
    }
}

void swap(int *a, int *b) {
    int temp;
    temp = *a;
    *a = *b;
    *b = temp;
}

File: algorithm/Sorting/bubble_sort.py
def bubble_sort(data_list):
    for index in range(len(data_list) - 1):
        swap = False
        for index2 in range(len(data_list) - index - 1):
            if data_list[index2] > data_list[index2 + 1]:
                data_list[index2], data_list[index2 + 1] = data_list[index2 + 1], data_list[index2]
                swap = True
        
        if swap == False:
            break
    return data_list